backend/app/main.py

```
@@ -479,153 +461,84 @@ async def get_mcp_logs():
-        }
-        logger.info(f"Returning error response: {error_response}")
-        return error_response
-
+        logger.error(f"Error initiating discovery for {request.url}: {str(e)}", exc_info=True)
+        # Return a structured error response immediately if initiation fails
+        # Note: Background task errors need separate handling within the task itself.
+        raise HTTPException(status_code=500, detail=f"Error initiating discovery: {str(e)}")
 @app.post("/api/crawl")
-async def crawl_endpoint(request: CrawlRequest):
-    """Crawl the provided pages and generate markdown content"""
+async def crawl_endpoint(request: CrawlRequest, background_tasks: BackgroundTasks):
```

Ruff (0.8.2)

Missing return type annotation for public function `crawl_endpoint`

(ANN201)